src/modules/Notifies/NotifyExtension.cpp

```diff
@@ -0,0 +1,119 @@
+	it under the terms of the GNU Lesser General Public License as published
+	by the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU Lesser General Public License for more details.
+
+	You should have received a copy of the GNU Lesser General Public License
+	along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <NotifyExtension.hpp>
+#include <Notify.hpp>
+
+#ifdef Q_OS_LINUX
+	#include <X11_Notify.hpp>
+#endif
+#include <TrayNotify.hpp>
+
+#include <QCoreApplication>
+
+NotifyService::NotifyService(Notify *notify, Settings &settings) :
+	notify(notify)
+{
+	Q_ASSERT(notify);
+	if (settings.getBool("show_title"))
+		connect(&QMPlay2Core, SIGNAL(updatePlaying(bool, const QString &, const QString &, const QString &, int, bool, const QString &)), this, SLOT(updatePlaying(bool, const QString &, const QString &, const QString &, int, bool, const QString &)));
+	if (settings.getBool("show_playState"))
+		connect(&QMPlay2Core, SIGNAL(playStateChanged(const QString &)), this, SLOT(playStateChanged(const QString &)));
+	if (settings.getBool("show_volume"))
+		connect(&QMPlay2Core, SIGNAL(volumeChanged(double)), this, SLOT(volumeChanged(double)));
+
+	if(settings.getBool("custom_msg"))
+	{
+		summaryFormat = settings.getString("custom_summary");
+		bodyFormat = settings.getString("custom_body");
+	}
+}
+
+NotifyService::~NotifyService()
+{
+	delete notify;
+}
+
+void NotifyService::volumeChanged(double v)
+{
+	notify->showMessage(tr("Volume Changed"), tr("Volume: %1%").arg((int)(100 * v)));
+}
+
+void NotifyService::updatePlaying(bool play, const QString &title, const QString &artist, const QString &album, int, bool, const QString &fileName)
+{
+	if(!play)
+		return;
+
+	QString summary = summaryFormat;
+	if(summary.isEmpty())
+	{
+		summary = title;
+		if (!title.isEmpty() && !artist.isEmpty())
+			summary += " - ";
+		summary += artist;
+	}
+	else
+		summary.replace("%title%", title).replace("%artist%", artist).replace("%album%", album).replace("%filename%", fileName);
+
+	if (summary.isEmpty())
+		summary = fileName;
+
+	QString body = bodyFormat;
+	if(!body.isEmpty())
+		body.replace("%title%", title).replace("%artist%", artist).replace("%album%", album).replace("%filename%", fileName);
+
+	if (body.isEmpty())
+		body = album;
+
+	notify->showMessage(summary, body);
+}
+
+void NotifyService::playStateChanged(const QString &playState)
+{
+	if (playState != "Playing")
+		notify->showMessage(QCoreApplication::applicationName(), playState);
+}
+
+NotifyExtension::NotifyExtension(Module &module) : notify(NULL)
+{
+	SetModule(module);
+}
+
+bool NotifyExtension::set()
+{
+	if (sets().getBool("type_disabled"))
+		delete this->notify;
```

Comment:
I think we should avoid `this->`. Also for new code please use `m_` prefix for private/protected class members :)